src/Features/CSharp/Portable/ExtractMethod/CSharpExtractMethodService.cs

```diff
@@ -2,31 +2,29 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
-#nullable disable
-
 using System.Composition;
 using System.Diagnostics.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
 using Microsoft.CodeAnalysis.ExtractMethod;
 using Microsoft.CodeAnalysis.Host.Mef;
-using Microsoft.CodeAnalysis.Options;
 using Microsoft.CodeAnalysis.Text;
 
-namespace Microsoft.CodeAnalysis.CSharp.ExtractMethod
-{
-    [Export(typeof(IExtractMethodService)), Shared]
-    [ExportLanguageService(typeof(IExtractMethodService), LanguageNames.CSharp)]
-    internal sealed class CSharpExtractMethodService : AbstractExtractMethodService<CSharpSelectionValidator, CSharpMethodExtractor, CSharpSelectionResult>
-    {
-        [ImportingConstructor]
-        [SuppressMessage("RoslynDiagnosticsReliability", "RS0033:Importing constructor should be [Obsolete]", Justification = "Used in test code: https://github.com/dotnet/roslyn/issues/42814")]
-        public CSharpExtractMethodService()
-        {
-        }
+namespace Microsoft.CodeAnalysis.CSharp.ExtractMethod;
 
-        protected override CSharpSelectionValidator CreateSelectionValidator(SemanticDocument document, TextSpan textSpan, ExtractMethodOptions options, bool localFunction)
-            => new(document, textSpan, options, localFunction);
+[Export(typeof(IExtractMethodService)), Shared]
+[ExportLanguageService(typeof(IExtractMethodService), LanguageNames.CSharp)]
+[method: ImportingConstructor]
+[method: SuppressMessage("RoslynDiagnosticsReliability", "RS0033:Importing constructor should be [Obsolete]", Justification = "Used in test code: https://github.com/dotnet/roslyn/issues/42814")]
+internal sealed class CSharpExtractMethodService() : AbstractExtractMethodService<
+    CSharpSelectionValidator,
+    CSharpMethodExtractor,
+    CSharpSelectionResult,
+    StatementSyntax,
+    ExpressionSyntax>
```

Comment:
introduced more strong types in the hierarchy here.  this allowed code in subclasses to move to base class while still being strongly typed.